scenarios/throughput_stress.go

```diff
@@ -140,13 +134,16 @@ func (t *tpsExecutor) Run(ctx context.Context, info loadgen.ScenarioInfo) error
 				})
 
 			if err == nil {
-				// The 1 is for the final workflow run.
-				curTotal := t.workflowCount.Add(uint64(result.TimesContinued + result.ChildrenSpawned + 1))
-
-				info.StatusCallback(ThroughputStressScenarioStatusUpdate{
-					CompletedIteration: curIteration,
-					CompletedWorkflows: curTotal,
-				})
+				t.updateState(result, curIteration)
+
+				if info.StatusCallback != nil {
+					stateJson, err := json.Marshal(t.state)
```

Comment:
Since we need to unmarshall (scenario option values are strings), it seems reasonable to also marshal it here. That way the client can treat the state as an opaque string.